Wilab-io/sample-apps
# [Deploy] Add the deployment functionality

# Description

In this PR, I added the ability to deploy the application once the required steps are ready.

# How to test

1. Go to your application's console: https://console.vespa-cloud.com/tenant/{tenant_name}/application/{app_name}/prod/deploy
and download the app key certificate into this app's `application` directory.

<img width="745" alt="image" src="https://github.com/user-attachments/assets/58dadf8c-4a95-4ef5-a792-256820bf2b5b">
<img width="346" alt="image" src="https://github.com/user-attachments/assets/206be000-5cf9-492e-aea9-8dcecafec61e">

2. Then go to the `src` directory and run `python main.py`.
3. When the app is up, you can log in with admin:1 or demo:1 users.
4. Upload at least one document.
5. Go through the configuration pages and fill in every required field.
6. The deploy button should now be activated, click it and follow the extra login step when prompted before hitting the "Continue" button.
7. Once you see the "Deployment successful" modal you can start searching your documents in the home page.

Closes #36
Closes #37
Closes #43
